app-creator/client/src/widgets/templates-tab/templates-tab.ts

```diff
@@ -85,15 +115,22 @@ export class TemplatesTab extends connect(store)(LitElement) {
    * Returns widgets with names and ids that include the search query.
    */
   static filterTemplates(
-    templateCards: { id: string; name: string; markup: TemplateResult }[],
-    query: string
+    templateCards: TemplatesTabItem[],
+    query: string,
+    deviceFilter: DeviceType
   ): Array<TemplatesTabItem> {
-    return templateCards.filter(({ id, name }) => {
+    return templateCards.filter(({ id, name, device }) => {
       const lowerCasedQuery = query.toLowerCase();
```

Comment:
The 'i' flag added to the `RegExp` makes the search case insensitive, so you don't have to lowercase the query.
You also don't need the `.*` since you didn't include the start and end characters `^` and `$`. Was there a particular case that was failing without it?